pyproject.toml

```diff
@@ -128,6 +141,10 @@ docker-compose = { version = "^1.28", optional = true }
 [tool.poetry.extras]
 server = ["fastapi", "uvicorn", "python-multipart", "python-jose", "fastapi-utils"]
 templates = ["copier", "jinja2-time", "black", "ruff"]
+secrets-aws = ["boto3"]
+secrets-gcp = ["google-cloud-secret-manager"]
+secrets-azure = ["azure-identity", "azure-keyvault-secrets"]
+secrets-hashicorp = ["hashicorp"]
```

Comment:
Shouldn't this be `hvac`, the dependency specified above?

Reply:
I was counting on pip install to catch something like this, but apparently I can put whatever I want here and it will just ignore it.

Reply:
This is really weird, it works on my machine in its current form:

```
pip install -e .[secrets-hashicorp]

...

  Attempting uninstall: zenml
    Found existing installation: zenml 0.34.0
    Uninstalling zenml-0.34.0:
      Successfully uninstalled zenml-0.34.0
Successfully installed hvac-1.0.2 zenml-0.34.0
```

I can't explain this.

Reply:
Ok, apparently, if you have optional packages in your pyproject.toml that are not tied to any "extras", they are installed by default when you ask for any extra, real or fake.